src/catalog/pgduckdb_catalog.cpp

```diff
@@ -0,0 +1,126 @@
+#include "pgduckdb/catalog/pgduckdb_catalog.hpp"
+#include "duckdb/parser/parsed_data/attach_info.hpp"
+#include "duckdb/parser/parsed_data/create_schema_info.hpp"
+#include "pgduckdb/catalog/pgduckdb_storage.hpp"
+#include "pgduckdb/catalog/pgduckdb_transaction.hpp"
+
+extern "C" {
+#include "postgres.h"
+#include "utils/fmgroids.h"
+#include "fmgr.h"
+#include "catalog/pg_namespace.h"
+#include "utils/syscache.h"
+#include "utils/builtins.h"
+#include "utils/rel.h"
+#include "utils/snapshot.h"
+#include "access/heapam.h"
+#include "access/htup_details.h"
+#include "catalog/indexing.h"
+#include "access/genam.h"
+#include "access/xact.h"
+}
+
+namespace duckdb {
+
+PostgresCatalog::PostgresCatalog(AttachedDatabase &db, const string &connection_string, AccessMode access_mode,
+                                 Snapshot snapshot, PlannerInfo *planner_info)
+    : Catalog(db), path(connection_string), access_mode(access_mode), snapshot(snapshot), planner_info(planner_info) {
+}
+
+unique_ptr<Catalog>
+PostgresCatalog::Attach(StorageExtensionInfo *storage_info_p, ClientContext &context, AttachedDatabase &db,
+                        const string &name, AttachInfo &info, AccessMode access_mode) {
+	string connection_string = info.path;
+
+	if (!storage_info_p) {
+		throw InternalException("PostgresCatalog should always have access to the PostgresStorageExtensionInfo");
+	}
+	auto &storage_info = static_cast<PostgresStorageExtensionInfo &>(*storage_info_p);
+	auto snapshot = storage_info.snapshot;
+	auto planner_info = storage_info.planner_info;
+	return make_uniq<PostgresCatalog>(db, connection_string, access_mode, snapshot, planner_info);
+}
+
+// ------------------ Catalog API ---------------------
+
+void
+PostgresCatalog::Initialize(bool load_builtin) {
+	return;
+}
+
+string
+PostgresCatalog::GetCatalogType() {
+	return "pgduckdb";
+}
+
+optional_ptr<CatalogEntry>
+PostgresCatalog::CreateSchema(CatalogTransaction transaction, CreateSchemaInfo &info) {
+	throw NotImplementedException("CreateSchema not supported yet");
+}
+
+optional_ptr<SchemaCatalogEntry>
+PostgresCatalog::GetSchema(CatalogTransaction transaction, const string &schema_name, OnEntryNotFound if_not_found,
+                           QueryErrorContext error_context) {
+	if (schema_name == DEFAULT_SCHEMA) {
+		return GetSchema(transaction, "public", if_not_found, error_context);
+	}
```

Comment:
Let's add a TODO comment here to make this work with the postgres search_path instead. Having `main` remapped to `public` only works when the Postgres search_path is unmodified.

Reply:
I think we can remove this if statement now that we have send fully qualified table references to duckdb.